modules/javafx.graphics/src/main/java/com/sun/javafx/scene/layout/HeaderButtonBehavior.java

```diff
@@ -54,20 +58,24 @@ public HeaderButtonBehavior(Node node, HeaderButtonType type) {
             .flatMap(Scene::windowProperty)
             .map(w -> w instanceof Stage s ? s : null);
 
-        if (type == HeaderButtonType.MAXIMIZE) {
-            subscription = Subscription.combine(
+        Subscription subscription = Subscription.combine(
+            stage.flatMap(Stage::fullScreenProperty).subscribe(this::onFullScreenChanged),
+            () -> node.removeEventHandler(MouseEvent.MOUSE_RELEASED, this));
+
+        if (type != HeaderButtonType.CLOSE) {
+            subscription = Subscription.combine(subscription,
+                stage.subscribe(this::onStageChanged),
                 stage.flatMap(Stage::resizableProperty).subscribe(this::onResizableChanged),
-                stage.flatMap(Stage::fullScreenProperty).subscribe(this::onFullScreenChanged),
-                stage.flatMap(Stage::maximizedProperty).subscribe(this::onMaximizedChanged),
-                () -> node.removeEventHandler(MouseEvent.MOUSE_RELEASED, this)
-            );
-        } else {
-            subscription = Subscription.combine(
-                stage.flatMap(Stage::fullScreenProperty).subscribe(this::onFullScreenChanged),
-                () -> node.removeEventHandler(MouseEvent.MOUSE_RELEASED, this)
-            );
+                () -> { if (getStage() instanceof Stage s) s.removeEventFilter(WindowEvent.WINDOW_SHOWING, this); });
+        }
+
+        if (type == HeaderButtonType.MAXIMIZE) {
+            subscription = Subscription.combine(subscription,
```

Comment:
here and on L66 the `subscription` pointer (set in L61) gets overwritten.  does it mean some subscription(s) won't be cancelled in `dispose()`?